sigstore/_verify.py

```diff
@@ -130,7 +135,13 @@ def verify(
     store.add_cert(openssl_intermediate)
     store.set_time(sign_date)
     store_ctx = X509StoreContext(store, openssl_cert)
-    store_ctx.verify_certificate()
+    try:
+        store_ctx.verify_certificate()
+    except X509StoreContextError as store_ctx_error:
+        return VerificationFailure(
+            reason="Failed to verify signing certificate, consider upgrading `sigstore` if a newer "
+            f"version is available: {store_ctx_error}"
+        )
```

Comment:
LGTM. I think we'll want to think a bit more about how we want to render exceptions in general, but we can hold off on that.